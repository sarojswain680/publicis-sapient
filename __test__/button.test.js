import '@testing-library/jest-dom/extend-expect';
import { cleanup, render } from '@testing-library/react';
import React from 'react';
import ReactDom from 'react-dom';
import renderer from "react-test-renderer";
import Button from '../src/components/Modules/button';

afterEach(cleanup)

it("renderers without crashing", () => {
    const div = document.createElement("div");
    ReactDom.render(<Button></Button>,div);
})

it("renders button correctly", () => {
const {getByTestId} =   render(<Button value="2006"></Button>)
expect(getByTestId('button').textContent).toBe("2006")
})

it("matches snapshot" , ()=> {
  const tree =  renderer.create(<Button value="2006"></Button>).toJSON();

  expect(tree).toMatchSnapshot();
})