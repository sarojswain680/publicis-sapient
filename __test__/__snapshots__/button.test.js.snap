// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained button"
    data-testid="button"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      2006
    </span>
  </button>
</div>
`;
